test/canary-environment/mzcompose.py

```diff
@@ -30,37 +30,86 @@
+
+            > CREATE CONNECTION IF NOT EXISTS kafka_connection TO KAFKA (
+              BROKER '{KAFKA_BOOTSTRAP_SERVER}',
+              SASL MECHANISMS = 'PLAIN',
+              SASL USERNAME = SECRET kafka_username,
+              SASL PASSWORD = SECRET kafka_password
+              )
+
+            > CREATE SECRET IF NOT EXISTS csr_username AS '{CONFLUENT_CLOUD_FIELDENG_CSR_USERNAME}'
+            > CREATE SECRET IF NOT EXISTS csr_password AS '{CONFLUENT_CLOUD_FIELDENG_CSR_PASSWORD}'
+
+            > CREATE CONNECTION IF NOT EXISTS csr_connection TO CONFLUENT SCHEMA REGISTRY (
+              URL '{SCHEMA_REGISTRY_ENDPOINT}',
+              USERNAME = SECRET csr_username,
+              PASSWORD = SECRET csr_password
+              )
+        """
+        )
+    )
 
+    c.testdrive(
+        input=dedent(
+            f"""
+            $ postgres-execute connection=postgres://postgres:postgres@postgres
+            ALTER USER postgres WITH replication;
 
-        > CREATE CONNECTION IF NOT EXISTS kafka_connection TO KAFKA (
-          BROKER '{KAFKA_BOOTSTRAP_SERVER}',
-          SASL MECHANISMS = 'PLAIN',
-          SASL USERNAME = SECRET kafka_username,
-          SASL PASSWORD = SECRET kafka_password
-          )
+            DROP SCHEMA IF EXISTS public CASCADE;
+            CREATE SCHEMA public;
 
-        > CREATE SECRET IF NOT EXISTS csr_username AS '{CONFLUENT_CLOUD_FIELDENG_CSR_USERNAME}'
-        > CREATE SECRET IF NOT EXISTS csr_password AS '{CONFLUENT_CLOUD_FIELDENG_CSR_PASSWORD}'
+            CREATE TABLE IF NOT EXISTS people (id INTEGER PRIMARY KEY, name TEXT DEFAULT REPEAT('a', 16), incarnation INTEGER DEFAULT 1);
+            ALTER TABLE people REPLICA IDENTITY FULL;
 
-        > CREATE CONNECTION IF NOT EXISTS csr_connection TO CONFLUENT SCHEMA REGISTRY (
-          URL '{SCHEMA_REGISTRY_ENDPOINT}',
-          USERNAME = SECRET csr_username,
-          PASSWORD = SECRET csr_password
-          )
-        """
+            CREATE TABLE IF NOT EXISTS relationships (a INTEGER, b INTEGER, incarnation INTEGER DEFAULT 1, PRIMARY KEY (a,b));
+            ALTER TABLE relationships REPLICA IDENTITY FULL;
+
+            DROP PUBLICATION IF EXISTS mz_source;
+            CREATE PUBLICATION mz_source FOR ALL TABLES;
+
+            CREATE EXTENSION IF NOT EXISTS pg_cron;
+
+            SELECT cron.schedule('insert-people', '1 seconds', 'INSERT INTO people (id) SELECT FLOOR(RANDOM() * {POSTGRES_RANGE}) FROM generate_series(1,2) ON CONFLICT (id) DO UPDATE SET incarnation = people.incarnation + 1');
+            SELECT cron.schedule('update-people-name', '1 seconds', 'UPDATE people SET name = REPEAT(id::text, 16) WHERE id = {POSTGRES_RANGE_FUNCTION}');
+            SELECT cron.schedule('update-people-incarnation', '1 seconds', 'UPDATE people SET incarnation = incarnation + 1 WHERE id = {POSTGRES_RANGE_FUNCTION}');
+            SELECT cron.schedule('delete-people', '1 seconds', 'DELETE FROM people WHERE id = {POSTGRES_RANGE_FUNCTION}');
```

Comment:
Are these statements not prefixed with `> ` because we don't want to compare the results because this is executed against Postgres?

Reply:
All of those statements are part of the `$ postgres-execute` directive and are run against Postgres. `$ postgres-execute` does not allow you to provide an "expected result" it just fails if any of the individual statements fail.